cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -662,16 +678,96 @@ func (c *client) WaitForResourceStatuses() error {
 				klog.V(10).Info("retrying: machine status is empty")
 				return false, nil
 			}
-			// if m.Status.ProviderStatus == nil {
-			// 	klog.V(10).Info("retrying: machine.Status.ProviderStatus is not set")
-			// 	return false, nil
-			// }
 		}
 
 		return true, nil
 	})
 }
 
+func (c *client) GetClusterSecrets(cluster *clusterv1.Cluster) ([]*corev1.Secret, error) {
+	list := &corev1.SecretList{}
+	if err := c.clientSet.List(ctx, list, ctrlclient.InNamespace(cluster.Namespace)); err != nil {
+		return nil, errors.Wrapf(err, "error listing Secrets for Cluster %s/%s", cluster.Namespace, cluster.Name)
+	}
+
+	res := []*corev1.Secret{}
+	for i, secret := range list.Items {
+		if strings.HasPrefix(secret.Name, cluster.Name) {
+			res = append(res, &list.Items[i])
+		}
+	}
+	return res, nil
+}
+
+func (c *client) CreateSecret(s *corev1.Secret) error {
+	if err := c.clientSet.Create(context.Background(), s); err != nil {
+		return errors.Wrapf(err, "error creating Secret %s/%s", s.Namespace, s.Name)
+	}
+	return nil
+}
+
+func (c *client) ForceDeleteSecret(namespace, name string) error {
```

Comment:
Are we actually setting finalizers anywhere for secrets? I don't think there is any reason that we should.

Reply:
That's fair, I figured a new controller might set these and left it in just in case. Would you prefer me to remove it?

Reply:
Yes, lets avoid making assumptions, especially if there is some controller that is setting them in the future, we wouldn't necessarily know how to migrate it

Reply:
Fixed